### DISCUSSION TITLE
Request for feedback: Snowflake destination - Introducing option to use `MERGE` for deduplication of final tables

### DISCUSSION THREAD

**Discussion by gisripa** [airbytehq/airbyte#44478] [Connector Ideas and Features]
Currently in Snowflake destination connector version < 3.11.9, the way we deduplicate final table is by doing an Insert + Delete + Delete (deletes captured using change data capture).  
We are introducing a feature to opt-in using connector configuration flag, which switches to using snowflake's `MERGE` statement for deduplication in final tables.  We would like to hear from the community. Give us a thumbs up if `MERGE` is performing better, or thumbs down if it isn't. Please also provide details of the schema, the nature of the updates on each sync and any other details which can help us replicate the setup. 
[PR Reference](https://github.com/airbytehq/airbyte/pull/43367)

### Background
We have received feedback from few customers that `MERGE` could out perform than how we do today. However when we released this same change earlier few customers reported that there was an increased warehouse activity after they upgraded the connector when `MERGE` statement was introduced, which caused us to rollback. We are rolling it out as an opt-in feature to hear more from our community.

**Comment by et-dynamic** [comment]
I tried this out at the suggestion of Airbyte support for a Postgres->Snowflake connection. The comparison wasn't too sophisticated; I just enabled the option for four days, and compared Snowflake costs (Airbyte warehouse only) before, during, and after those four days. The result was that average daily costs for the two periods with the option disabled were both almost identical (within 1% of one another), while the period with the option enabled had about 25% higher average daily costs than the others.

Would be happy to provide more details via DM.

[a comment by an automated account is not shown]
